pkg/executor/executor_test.go

```diff
@@ -0,0 +1,361 @@
+/*
+Copyright 2022 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package executor
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+)
+
+func TestNormalJobs(t *testing.T) {
+	logrus.SetLevel(logrus.TraceLevel)
+	a := assert.New(t)
+	c := Config{
+		QueueSize:            5,
+		Workers:              3,
+		MaxJobRetries:        0,
+		BaseRetryDelay:       10 * time.Millisecond,
+		RetryJobAfterFailure: false,
+		PerWorkerQPS:         5,
+		Timeout:              200 * time.Millisecond,
+	}
+
+	e, err := New(c)
+	a.NoError(err)
+	a.NoError(waitForAdded(e.queue, 0))
+
+	err = e.AddJob(&sleepingJob{100 * time.Millisecond, "1"})
```

Comment:
Maybe it's better to use a for loop to add these jobs. Same below.

Reply:
good. improved using a for loop